cli/ops/process.rs

```diff
@@ -214,14 +188,17 @@ fn op_run_status(
   let rid = args.rid as u32;
 
   state.check_run()?;
-
-  let future = ChildStatus {
-    rid,
-    state: state.clone(),
-  };
+  let state = state.clone();
 
   let future = async move {
-    let run_status = future.await?;
+    let run_status = {
+      let mut table = state.lock_resource_table();
+      let child_resource = table
+        .get_mut::<ChildResource>(rid)
+        .ok_or_else(bad_resource)?;
+      (&mut child_resource.child).await.map_err(ErrBox::from)?
+    };
```

Comment:
Holds resource table mutex until `await` resolves - that's bad

Reply:
@ry this statement still hold true, I suggest putting `child_resource.child` behind a mutex and cloning it - that way resource table won't be locked.

Reply:
I've undone all the changes to this module - turns out to be unrelated.